pkg/aws/aws.go

```diff
@@ -0,0 +1,93 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+
+	"k8s.io/client-go/rest"
+)
+
+const (
+	messageRetrieveVPCID      = "Retrieving VPC ID"
+	messageRetrievedVPCID     = "Retrieved VPC ID %s"
+	messagePortOpen           = "Opening port %v protocol %s for intra-cluster communications"
+	messageCreateGatewaySG    = "Creating Submariner gateway security group"
+	messageCreatedGatewaySG   = "Created Submariner gateway security group %s"
+	messageAdjustPublicSubnet = "Adjusting public subnet %s to support Submariner"
+	messageDeployGatewayNode  = "Deploying gateway node"
+)
+
+type awsCloud struct {
+	client         ec2iface.EC2API
+	k8sConfig      *rest.Config
+	gwInstanceType string
+	infraID        string
+	region         string
+}
+
+// NewAwsCloud creates a new api.Cloud instance which can prepare AWS for Submariner to be deployed on it
+func NewAwsCloud(k8sConfig *rest.Config, awsSession *session.Session, infraID, gwInstanceType string) (api.Cloud, error) {
```

Comment:
Avoids "stutter" since `aws` is the package name:

```suggestion
func NewCloud(k8sConfig *rest.Config, awsSession *session.Session, infraID, gwInstanceType string) (api.Cloud, error) {
```